app/views/feeds/_form_apple_config.html.erb

```diff
@@ -3,28 +3,31 @@
     <div class="card-header card-header-light d-flex justify-content-between">
       <h2 class="card-title h5 flex-grow-1"><%= t(".title") %></h2>
     </div>
-    <div class="card-body">
+    <div class="card-body" data-controller="apple-key">
       <div class="mb-4">
         <p><%= t(".description") %></p>
         <a href="https://help.prx.org"><%= t(".guide_link") %></a>
       </div>
       <%= form.fields_for :apple_config do |config_form| %>
         <%= config_form.fields_for :key do |key_fields| %>
-          <div class="mb-4">
-            <div class="form-floating">
-              <%= key_fields.text_field :provider_id, disabled: config_form.object.persisted? %>
-              <%= key_fields.label :provider_id %>
-            </div>
+          <% unless config_form.object.persisted? %>
+            <%= key_fields.file_field :key_file, class: "mb-4 form-control", accept: ".p8, .pem", data: {action: "apple-key#convertFileToKey"} %>
+          <% end %>
```

Comment:
i opted to leave the disabled fields showing at all times. i was having trouble getting them to conditionally pop up for the error and this was the cleanest i could get to work properly.

Reply:
+1 works for me.